components/brave_shields/core/browser/brave_shields_settings.cc

```diff
@@ -0,0 +1,170 @@
+  CHECK(host_content_settings_map_);
+}
+
+BraveShieldsSettings::~BraveShieldsSettings() = default;
+
+void BraveShieldsSettings::SetBraveShieldsEnabled(bool is_enabled,
+                                                  const GURL& url) {
+  brave_shields::SetBraveShieldsEnabled(host_content_settings_map_, is_enabled,
+                                        url, local_state_);
+}
+
+bool BraveShieldsSettings::GetBraveShieldsEnabled(const GURL& url) {
+  return brave_shields::GetBraveShieldsEnabled(host_content_settings_map_, url);
+}
+
+void BraveShieldsSettings::SetDefaultAdBlockMode(mojom::AdBlockMode mode) {
+  SetAdBlockMode(mode, GURL());
+}
+
+mojom::AdBlockMode BraveShieldsSettings::GetDefaultAdBlockMode() {
+  return GetAdBlockMode(GURL());
+}
+
+void BraveShieldsSettings::SetAdBlockMode(mojom::AdBlockMode mode,
+                                          const GURL& url) {
+  ControlType control_type_ad;
+  ControlType control_type_cosmetic;
+
+  if (mode == mojom::AdBlockMode::ALLOW) {
+    control_type_ad = ControlType::ALLOW;
+  } else {
+    control_type_ad = ControlType::BLOCK;
+  }
+
+  if (mode == mojom::AdBlockMode::AGGRESSIVE) {
+    control_type_cosmetic = ControlType::BLOCK;  // aggressive
+  } else if (mode == mojom::AdBlockMode::STANDARD) {
+    control_type_cosmetic = ControlType::BLOCK_THIRD_PARTY;  // standard
+  } else {
+    control_type_cosmetic = ControlType::ALLOW;  // allow
+  }
+
+  brave_shields::SetAdControlType(host_content_settings_map_, control_type_ad,
+                                  url, local_state_);
+
+  brave_shields::SetCosmeticFilteringControlType(host_content_settings_map_,
+                                                 control_type_cosmetic, url,
+                                                 local_state_, profile_state_);
+}
+
+mojom::AdBlockMode BraveShieldsSettings::GetAdBlockMode(const GURL& url) {
+  ControlType control_type_ad =
+      brave_shields::GetAdControlType(host_content_settings_map_, url);
+
+  ControlType control_type_cosmetic =
+      brave_shields::GetCosmeticFilteringControlType(host_content_settings_map_,
+                                                     url);
+
+  if (control_type_ad == ControlType::ALLOW) {
+    return mojom::AdBlockMode::ALLOW;
+  }
+
+  if (control_type_cosmetic == ControlType::BLOCK) {
+    return mojom::AdBlockMode::AGGRESSIVE;
+  } else {
+    return mojom::AdBlockMode::STANDARD;
+  }
+}
+
+void BraveShieldsSettings::SetDefaultFingerprintMode(
+    mojom::FingerprintMode mode) {
+  SetFingerprintMode(mode, GURL());
+}
+
+mojom::FingerprintMode BraveShieldsSettings::GetDefaultFingerprintMode() {
+  return GetFingerprintMode(GURL());
+}
+
+void BraveShieldsSettings::SetFingerprintMode(mojom::FingerprintMode mode,
+                                              const GURL& url) {
+#if BUILDFLAG(IS_IOS)
+  /// Strict FingerprintMode is not supported on iOS
+  CHECK(mode != mojom::FingerprintMode::STRICT_MODE);
```

Comment:
May want to use a `DCHECK` here as even when its set to `STRICT_MODE` its not necessarily fatal as below you will map it back to standard

Reply:
https://github.com/brave/brave-core/pull/30033#discussion_r2261106239